src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -854,34 +822,75 @@ public boolean shouldSkipMonitoredTripCheck(boolean persist) throws Exception {
-                // TODO: Change log level.
-                LOG.info("Trip happening soon. Checking.");
+                LOG.debug("Trip happening soon. Checking.");
                 return false;
             }
             // If the trip starts within 30 minutes, check the trip every minute (assuming the loop runs every minute).
             int checkEveryMinuteThresholdMinutes = 30;
             if (minutesUntilTrip <= checkEveryMinuteThresholdMinutes) {
-                // TODO: Change log level.
-                LOG.info("Trip happening within {} minutes. Checking every minute.", checkEveryMinuteThresholdMinutes);
+                LOG.debug("Trip happening within {} minutes. Checking every minute.", checkEveryMinuteThresholdMinutes);
                 return false;
             }
         }
         // TODO: Check that journey state is not flagged
         // TODO: Check last notification time.
         // Default to skipping check.
-        // TODO: Change log level.
-        LOG.info("Trip criteria not met to check. Skipping.");
+        LOG.debug("Trip criteria not met to check. Skipping.");
         return true;
     }
 
+    /**
+     * Calculate target time for the next trip plan request find the next possible day the trip is active by
+     * initializing the appropriate target time. Start by checking today's date at the earliest in case the user has
+     * paused trip monitoring for a while.
+     */
+    private void setTargetDateTime() {
+        // Initialize to the current date with the trip's hours and minutes.
+        targetZonedDateTime = trip.tripZonedDateTime(DateTimeUtils.nowAsLocalDate());
+
+        // Attempt to advance to the next monitored day, except for one-time trips
+        // or if tracking is ongoing or if the matching itinerary is still valid.
+        if (shouldAdvanceToNextDay()) {
+            // Check if the journeyState indicates that an itinerary has already been calculated in a previous run of
+            // this CheckMonitoredTrip. If the targetDate is null, then the current date has not yet been checked. If
+            // the journeyState's targetDate is not null, that indicates that today has already been checked.
+            // Therefore, advance targetDate by another day before calculating when the next itinerary occurs.
+            if (previousJourneyState.targetDate != null) {
+                LocalDate lastDate = DateTimeUtils.getDateFromString(
+                    previousJourneyState.targetDate,
+                    DateTimeUtils.DEFAULT_DATE_FORMAT_PATTERN
+                );
+                if (
+                    lastDate.getYear() == targetZonedDateTime.getYear() &&
+                    lastDate.getMonthValue() == targetZonedDateTime.getMonthValue() &&
+                    lastDate.getDayOfMonth() == targetZonedDateTime.getDayOfMonth()
+                ) {
+                    targetZonedDateTime = targetZonedDateTime.plusDays(1);
```

Comment:
Trying to avoid setting intermediate values to that field. A local variable should be used for these intermediate assignments, here and at L856 above, and inside `advanceToNetActiveTripDate`.